### PR TITLE
README for Microsoft.Extensions.Diagnostics.Testing package

### PR DESCRIPTION
**Pull request by omsoftdev**
README file has been updated in the Microsoft.Extensions.Diagnostics.Testing package.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4697)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Microsoft.Extensions.Diagnostics.Testing
 
-Hand-crafted fakes to make telemetry-related testing easier.
+This library provides utilities for easy testing logging and metering functionality.
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:3]:
```suggestion
This library provides utilities for convenient testing of logging and metering functionality.
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 2]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Example
+
+### Logging fakes
+
+The `FakeLogger` is the implementation of the `Microsoft.Extensions.Logging.ILogger` that collects log messages
+in a memory list of log records. It is designed to be used for validation that the functionality under
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:26]:
```suggestion
in a list of log records. It is designed to be used for validation that the functionality under
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 3]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
...
+test writes the expected log messages:
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:27]:
```suggestion
test emits the expected log records:
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 4]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
...
+These extension methods register the `FakeLogCollector` - the in-memory holder of log messages
+and the `FakeLoggerProvider` - the implementation of `Microsoft.Extensions.Logging.ILoggerProvider` that returns `FakeLogger` instances.
+
+The following example shows the usage of fake logging in combination with the dependency injection container in a unit test:
+
+```csharp
+[Fact]
+public void TestMethod()
+{
+  var serviceProvider = new ServiceCollection()
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:59]:
```suggestion
  using var serviceProvider = new ServiceCollection()
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 5]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
...
+    .AddFakeLogging();
+    .AddSingleton<TheClassUnderTest>()
+    .BuildServiceProvider();
+
+  var classUnderTest = serviceProvider.GetRequiredService<TheClassUnderTest>();
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:64]:
```suggestion
  // The "ILogger" is injected by DI as a constructor parameter to "TheClassUnderTest":
  var classUnderTest = serviceProvider.GetRequiredService<TheClassUnderTest>();
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 6]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
 </ItemGroup>
 ```
 
+## Usage Example
+
+### Logging fakes
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:23]:
```suggestion
### Logs collection
```

File: src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md [hunk 7]
```diff
@@ -18,6 +18,84 @@ Or directly in the C# project file:
...
+
+The `MetricCollector` is a utility class that can be used to test that metrics are published correctly by a metering instrument:
+
+```csharp
+[Fact]
+public void TestMethod()
+{
+  using var meter = new Meter("MyMeter");
+  using var counterInstrument = meter.CreateCounter<long>("request.counter");
+  using var collector = new MetricCollector(meter, "request.counter");
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md:89]:
```suggestion
  using var collector = new MetricCollector<long>(meter, "request.counter");
```

Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Diagnostics.Testing/README.md]: @omsoftdev ?

**Review by xakep139 — APPROVED** [review]

**Comment by joperezr** [comment]
Re-running CI since the PR was re-targeted to main.

**Comment by joperezr** [comment]
In for 8.1. Thanks a lot @omsoftdev

**Pull request merged**